swarm.go

```diff
@@ -37,6 +37,18 @@ var ErrSwarmClosed = errors.New("swarm closed")
 // transport is misbehaving.
 var ErrAddrFiltered = errors.New("address filtered")
 
+// ErrNoConn is returned when attempting to open a stream to a peer with the NoDial
+// option and no usable connection is available.
+var ErrNoConn = errors.New("no usable connection to peer")
+
+// ContextOption is the type of context options understood by the swarm
+type ContextOption string
```

Comment:
Nit: This should probably be a newtype of `struct{}`.

Also, we should probably put this in go-libp2p-net. Now that we're using go modules, this should be _easy_ (:tada:)

Reply:
They need to be comparable, whatever that implies. Also the example in godoc uses a string alias type, so I went with it.

Reply:
I'll update go-libp2p-net to provide the option.

Reply:
Moved to `go-libp2p-net` in https://github.com/libp2p/go-libp2p-net/pull/42

Reply:
Updated to use the nodial option in go-libp2p-net.